utils/ls8usgsprepare.py

```diff
@@ -0,0 +1,177 @@
+              ('5', 'nir'),
+              ('6', 'swir1'),
+              ('7', 'swir2'),
+              ('8', 'panchromatic'),
+              ('9', 'cirrus'),
+              ('10', 'lwir1'),
+              ('11', 'lwir2'),
+              ('QUALITY', 'quality')]
+
+    return {
+        'id': str(uuid.uuid5(uuid.NAMESPACE_URL, path)),
+        'processing_level': level,
+        'product_type': product_type,
+        # 'creation_dt': ct_time,
+        'label': info['METADATA_FILE_INFO']['LANDSAT_SCENE_ID'],
+        'platform': {'code': info['PRODUCT_METADATA']['SPACECRAFT_ID']},
+        'instrument': {'name': info['PRODUCT_METADATA']['SENSOR_ID']},
+        # 'acquisition': {'groundstation': {'code': station}},
+        'extent': {
+            'from_dt': sensing_time,
+            'to_dt': sensing_time,
+            'center_dt': sensing_time,
+            'coord': get_coords(geo_ref_points, spatial_ref),
+        },
+        'format': {'name': info['PRODUCT_METADATA']['OUTPUT_FORMAT']},
+        'grid_spatial': {
+            'projection': {
+                'geo_ref_points': geo_ref_points,
+                'spatial_reference': 'EPSG:%s' % cs_code,
+                #     'valid_data': {
+                #         'coordinates': tileInfo['tileDataGeometry']['coordinates'],
+                #         'type': tileInfo['tileDataGeometry']['type']}
+            }
+        },
+        'image': {
+            'bands': {
+                image[1]: {
+                    'path': info['PRODUCT_METADATA']['FILE_NAME_BAND_' + image[0]],
+                    'layer': 1,
+                } for image in images
+            }
+        },
+        'L1_METADATA_FILE': info,
+        'lineage': {'source_datasets': {}},
+    }
+
+
+def absolutify_paths(doc, path):
+    for band in doc['image']['bands'].values():
+        band['path'] = os.path.join(path, band['path'])
+    return doc
+
+
+@click.command(help="Prepare USGS Landsat Collection 1 data for ingestion into the Data Cube."
+                    "To Set the Path for referring the datasets -Download the  Landsat scene data "
+                    " from Earth Explorer or GloVis into some_space_available_folder and unpack the file. "
+                    " For example: yourscript.py --output [Yaml- which writes datasets into this file for indexing] "
+                    "[Path for dataset as : /home/some_space_available_folder/]")
+@click.option('--output', help="Write datasets into this file",
+              type=click.Path(exists=False, writable=True, dir_okay=False))
+@click.argument('datasets',
+                nargs=-1)
+def main(output, datasets):
+    logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.INFO)
+
+    if output:
+        docs = (absolutify_paths(prepare_dataset(path), path) for path in datasets)
+        with open(output, 'w') as stream:
+            yaml.dump_all(docs, stream)
+    else:
+        raise RuntimeError('must specify --output')
```

Comment:
Setting `required=False` for `@click.option('--ouput`, ... ` will take care of the checking for you.